selfdrive/ui/qt/widgets/keyboard.cc

```diff
@@ -7,31 +7,31 @@
 
 #include "keyboard.hpp"
 
-KeyboardLayout::KeyboardLayout(QWidget* parent, std::vector<QVector<QString>> layout) : QWidget(parent) {
+KeyboardLayout::KeyboardLayout(QWidget *parent, std::vector<QVector<QString>> layout) : QWidget(parent) {
   QVBoxLayout* vlayout = new QVBoxLayout;
   QButtonGroup* btn_group = new QButtonGroup(this);
 
   QObject::connect(btn_group, SIGNAL(buttonClicked(QAbstractButton*)), parent, SLOT(handleButton(QAbstractButton*)));
 
   int i = 0;
-  for(auto s : layout){
+  for (auto s : layout){
     QHBoxLayout *hlayout = new QHBoxLayout;
 
     if (i == 1){
       hlayout->addSpacing(90);
     }
 
-    for(QString p : s){
+    for (QString p : s){
       QPushButton* btn = new QPushButton(p);
       btn->setFixedHeight(120);
-
+      btn_group->addButton(btn);
+      hlayout->addSpacing(10);
       if (p == QString("  ")){
-        btn->setFixedWidth(1024);
+        hlayout->addWidget(btn, 3);
```

Comment:
Do we want a comment for what the 3 and 1 do?

Reply:
It's fine.

Reply:
don’t think that’s necessary, but maybe move them to constants on top for the different stretch values